pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/SegmentGeneratorConfig.java

```diff
@@ -249,7 +250,13 @@ public SegmentGeneratorConfig(TableConfig tableConfig, Schema schema) {
   }
 
   public Map<String, Map<String, String>> getColumnProperties() {
-    return _columnProperties;
+    HashMap<String, Map<String, String>> copy = new HashMap<>();
```

Comment:
Making it immutable during read time can be quite expensive. We can make it immutable during construction time

Reply:
I've applied this change